core/tests/unit/ldml/keyboards/k_102_keytest.xml

```diff
@@ -1,19 +1,28 @@
 <?xml version="1.0" encoding="UTF-8"?>
 
 <!--
-@@keys: [K_BKQUOTE][K_1][K_ENTER][K_Z][K_Q][K_BKQUOTE][K_Z][K_Q][K_2][K_BKQUOTE][K_Z][K_Q][K_1][K_BKQUOTE][K_Z][K_Q][K_ENTER][K_BKQUOTE][K_Z]
-@@expected: \u0061\u007a\u0041\u007a\u0041\u007a\u0041\u007a\u0041\u007a
--->
+Note that the 'expected' lines are cumulative unless there's a reset event.
+
+@@keys:      [K_BKQUOTE][K_1][K_ENTER][K_Z]
+@@expected:  z
+Comment:     1=gap, enter=not mapped/mappable - reset. (z used as a subtest separator)
+
+@@keys:      [K_Q][K_BKQUOTE][K_Z]
+@@expected:  zAz
+Comment:     q\m{q}a => A due to transform
+
+@@keys:      [K_Q][K_2][K_BKQUOTE][K_Z]
+@@expected:  zAzAz
+Comment:     2=not mapped, but NO reset.
+
+@@keys:      [K_Q][K_1][K_BKQUOTE][K_Z]
+@@expected:  zAzAzAz
+Comment:     1 is a gap (no effect) so no ctx reset
+
+@@keys:      [K_Q][K_ENTER][K_BKQUOTE][K_Z]
+@@expected:  az
+Comment:     enter=not mappable, causes ctx reset.
```

Comment:
Making this comment not on this PR, but to say if this sequence was entered by the end user  when #10471 is implemented a context-aware app will restore the context after an enter and the transform will give the `A`.
For non-context away apps it will be as this test shows.